File: fixtures/csv-injection-node/vuln.js
// PLANTED VULNERABILITY (models openclaw's google-meet `csvCell`): RFC-4180 quote-escaping only —
// it wraps a cell containing `" , \n`, but does NO formula-prefix neutralization. An untrusted cell
// like `=cmd|'/c calc'!A1` contains none of those trigger chars, so it reaches the CSV verbatim and
// executes when the file is opened in a spreadsheet. The lane's canary drives this to prove itself live.
function csvCell(value) {
  const text = value == null ? "" : String(value);
  return /[",\n]/.test(text) ? `"${text.replaceAll('"', '""')}"` : text;
}

function toCsv(rows) {
  return rows.map((row) => row.map(csvCell).join(",")).join("\n");
}

module.exports.toCsv = toCsv;

File: fixtures/csv-injection-swift/vuln.swift
import Foundation

// PLANTED VULNERABILITY (do not "fix"): RFC-4180 quote-escaping only, with NO formula-prefix guard —
// a cell whose value begins with `= + - @` is emitted verbatim and executes when the CSV is opened in
// Excel/Sheets. Models the real openclaw google-meet `csvCell`. Exists so the Swift CSV/formula-
// injection lane's canary can prove itself LIVE by driving a `=<marker>` value that survives into the
// output un-neutralized. Runs on the macOS host.
func csvCell(_ value: String) -> String {
    var out = value
    if out.contains(",") || out.contains("\"") || out.contains("\n") {
        out = "\"" + out.replacingOccurrences(of: "\"", with: "\"\"") + "\""
    }
    return out + ",count"
}
